spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -200,18 +200,35 @@ public boolean rmdir(String directory) throws IOException {
 	public boolean exists(String path) throws IOException {
 		Assert.hasText(path, "'path' must not be empty");
 
-		String currentWorkingPath = this.client.printWorkingDirectory();
-		Assert.state(currentWorkingPath != null,
-				"working directory cannot be determined, therefore exists check can not be completed");
 		boolean exists = false;
-
+		InputStream inputStream = null;
 		try {
-			if (this.client.changeWorkingDirectory(path)) {
-				exists = true;
-			}
+			inputStream = this.client.retrieveFileStream(path);
```

Comment:
I don't like this solution - it's a lot of overhead on the server to get ready for a file transfer which we will never execute.

I think we need to use the previous technique from this PR for a file and fall back to the test below on if false is returned, in case the test was for a directory.